okhttp/src/main/java/okhttp3/Response.java

```diff
@@ -148,21 +148,16 @@ public Headers headers() {
    * applications should set a modest limit on {@code byteCount}, such as 1 MiB.
    */
   public ResponseBody peekBody(long byteCount) throws IOException {
-    BufferedSource source = body.source();
-    source.request(byteCount);
-    Buffer copy = source.buffer().clone();
-
-    // There may be more than byteCount bytes in source.buffer(). If there is, return a prefix.
-    Buffer result;
-    if (copy.size() > byteCount) {
-      result = new Buffer();
-      result.write(copy, byteCount);
-      copy.clear();
-    } else {
-      result = copy;
+    BufferedSource peeked = body.source().peek();
+    Buffer buffer = new Buffer();
+    while (byteCount > 0) {
```

Comment:
I think you can eliminate this loop with a call to `request(byteCount)` which will do it for you.